src/os/bluestore/BlueStore.cc

```diff
@@ -11521,34 +11521,43 @@ void BlueStore::BSPerfTracker::update_from_perfcounters(
 }
 
 #ifdef HAVE_LIBZBD
-// For every object we maintain <zone_num+oid, offset> tuple in the key-value
-// store.  When a new object written to a zone, we insert the corresponding
-// tuple to the database.  When an object is truncated, we remove the
-// corresponding tuple.  When an object is overwritten, we remove the old tuple
-// and insert a new tuple corresponding to the new location of the object.  The
-// cleaner can now identify live objects within the zone <zone_num> by
-// enumerating all the keys starting with <zone_num> prefix.
+// For every object we maintain <zone_num+oid, offset> tuple in the
+// PREFIX_ZONED_CL_INFO namespace.  When a new object written to a zone, we
+// insert the corresponding tuple to the database.  When an object is truncated,
+// we remove the corresponding tuple.  When an object is overwritten, we remove
+// the old tuple and insert a new tuple corresponding to the new location of the
+// object.  The cleaner can now identify live objects within the zone <zone_num>
+// by enumerating all the keys starting with <zone_num> prefix.
 void BlueStore::_zoned_update_cleaning_metadata(TransContext *txc) {
   for (const auto &[o, offsets] : txc->zoned_onode_to_offset_map) {
-    std::string key;
-    get_object_key(cct, o->oid, &key);
     for (auto offset : offsets) {
       if (offset > 0) {
 	bufferlist offset_bl;
 	encode(offset, offset_bl);
-        txc->t->set(_zoned_get_prefix(offset), key, offset_bl);
+        txc->t->set(PREFIX_ZONED_CL_INFO, 
+		    _zoned_key(offset / bdev->get_zone_size(), &o->oid), 
+		    offset_bl);
       } else {
-        txc->t->rmkey(_zoned_get_prefix(-offset), key);
+        txc->t->rmkey(PREFIX_ZONED_CL_INFO, 
+		      _zoned_key(-offset / bdev->get_zone_size(), &o->oid));
```

Comment:
My bad. Fixed it. I went back and forth between accepting an offset or the zone number for _zoned_key because at another place I only have the zone number and I have to multiply it by bdev->get_zone_size(), but eventually decided to take just the offset, and forgot to update it here.